### PR TITLE
Improve docs of log_softmax

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
This is PR for improving docs of functions/links. Related issue: #2182

Modified functions/links:

- functions/activation/log_softmax

### CODE REVIEW

File: chainer/functions/activation/log_softmax.py [hunk 1]
```diff
@@ -108,15 +107,29 @@ def log_softmax(x, use_cudnn=True):
...
 
+    .. admonition:: Example
+
+        >>> x = np.array([[0, 1, 2], [0, 2, 4]], 'f')
+        >>> x
+        array([[ 0.,  1.,  2.],
+               [ 0.,  2.,  4.]], dtype=float32)
+        >>> F.log_softmax(x).data
+        array([[-2.40760589, -1.40760589, -0.40760589],
+               [-4.14293146, -2.14293146, -0.14293146]], dtype=float32)
```
Comment (mitmul) [inline chainer/functions/activation/log_softmax.py]: How about showing the result of `F.log(F.softmax(x))` also in this example? For example,
```
>>> F.log(F.softmax(x)).data
array([[-2.40760589, -1.40760589, -0.40760598],
       [-4.14293146, -2.1429317 , -0.1429316 ]], dtype=float32)
```
or
```
>>> np.allclose(F.log_softmax(x).data, F.log(F.softmax(x)).data)
True
```
etc.

Comment (keisuke-umezawa) [inline chainer/functions/activation/log_softmax.py]: I fixed the example to add the codes below
```
>>> np.allclose(F.log_softmax(x).data, F.log(F.softmax(x)).data)
True
```

File: chainer/functions/activation/log_softmax.py [hunk 2]
```diff
@@ -108,15 +107,31 @@ def log_softmax(x, use_cudnn=True):
         ``log_softmax`` method is more stable.
 
     Args:
-        x (~chainer.Variable): Input variable.
-        use_cudnn (bool): If ``True``, cuDNN is enabled and cuDNN ver. 3 or
-            later is used, then this function uses cuDNN as the core
-            implementation.
+        X (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
```
Comment (mitmul) [inline chainer/functions/activation/log_softmax.py]: `X` should be lower-case because the input argument shown in the top line uses a lower-case letter.

File: chainer/functions/activation/log_softmax.py [hunk 3]
```diff
@@ -108,15 +107,31 @@ def log_softmax(x, use_cudnn=True):
...
+        :class:`cupy.ndarray`):
+            Input variable.
+            A :math:`n`-dimensional (:math:`n \\geq 2`) float array.
+        use_cudnn (bool): If ``True`` and cuDNN is enabled, then this function
+            uses cuDNN as the core implementation.
 
     Returns:
         ~chainer.Variable: Output variable.
+        A :math:`n`-dimensional (:math:`n \\geq 2`) float array, which is the
+        same shape with X.
```
Comment (mitmul) [inline chainer/functions/activation/log_softmax.py]: `X` should be lower-case because the input argument shown in the top line uses a lower-case letter.

File: chainer/functions/activation/log_softmax.py [hunk 4]
```diff
@@ -87,17 +87,16 @@ def backward(self, x, gy):
...
 
     This function computes its logarithm of softmax along the second axis. Let
-    :math:`x = (x_1, x_2, \\dots, x_d)^{\\top}` be the d dimensional index
-    array and :math:`f(x_1, \\dots, x_d)` be the corresponding input array.
-    For each index :math:`x` in the input array, it computes the logarithm
-    of the probability :math:`\log p(x)` defined as
+    :math:`x = (x_1, x_2, \\dots, x_D)^{\\top}` be the D dimensional input
+    array.
+    For each input array :math:`x`, it computes the logarithm of the function
+    :math:`\log f(x)` defined as
```
Comment (mitmul) [inline chainer/functions/activation/log_softmax.py]: "logarithm of the function log f(x)" looks duplicated expression. "logarithm of the function f(x)" is more clear. So how about replace this with :math:`f(x)` ?

File: chainer/functions/activation/log_softmax.py (hunk 3 above)
Comment (keisuke-umezawa) [inline chainer/functions/activation/log_softmax.py]: I checked doctest as following,

```
$ make doctest

...

Document: reference/functions
-----------------------------
1 items passed all tests:
  88 tests in default
88 tests in 1 items.
88 passed and 0 failed.
Test passed.

...

Doctest summary
===============
  309 tests
   10 failures in tests
    0 failures in setup code
    0 failures in cleanup code
build finished with problems.
make: *** [doctest] Error 1
```

**Comment by keisuke-umezawa** [comment]
I checked doctest as following,

```
$ make doctest

...

Document: reference/functions
-----------------------------
1 items passed all tests:
  88 tests in default
88 tests in 1 items.
88 passed and 0 failed.
Test passed.

...

Doctest summary
===============
  309 tests
   10 failures in tests
    0 failures in setup code
    0 failures in cleanup code
build finished with problems.
make: *** [doctest] Error 1
```

File: chainer/functions/activation/log_softmax.py [hunk 5]
```diff
@@ -87,17 +87,16 @@ def backward(self, x, gy):
...
 
     This function computes its logarithm of softmax along the second axis. Let
-    :math:`x = (x_1, x_2, \\dots, x_d)^{\\top}` be the d dimensional index
-    array and :math:`f(x_1, \\dots, x_d)` be the corresponding input array.
-    For each index :math:`x` in the input array, it computes the logarithm
-    of the probability :math:`\log p(x)` defined as
+    :math:`x = (x_1, x_2, \\dots, x_D)^{\\top}` be the D dimensional input
+    array.
+    For each input array :math:`x`, it computes the logarithm of the function
+    :math:`f(x)` defined as
```
Comment (delta2323) [inline chainer/functions/activation/log_softmax.py]: Comparing the description with input argument of this function, I felt that `x` in the document corresponds to `x` in the argument at first sight because it uses same variable name. But it is not true. Actually, `x` in the document is a narrowed version of `x` in the argument. How about changing the variable name of the two?

Comment (mitmul) [inline chainer/functions/activation/log_softmax.py]: @keisuke-umezawa Could you check?

Comment (keisuke-umezawa) [inline chainer/functions/activation/log_softmax.py]: ok. I will fix them.

**Comment by keisuke-umezawa** [comment]
@mitmul plz merge after the test.

File: chainer/functions/activation/log_softmax.py [hunk 6]
```diff
@@ -87,17 +87,15 @@ def backward(self, x, gy):
...
-    of the probability :math:`\log p(x)` defined as
+    This function computes its logarithm of softmax along the second axis.
+    Let :math:`c = (c_1, c_2, \\dots, c_D)` be the slice of ``x`` along with
+    the second axis. For each slice :math:`c`, it computes the logarithm of
+    the function :math:`f(c)` defined as
 
     .. math::
-        p(x) = {\\exp(f(x_1, x_2, \\dots, x_d))
-                \\over \\sum_{x'_2} \\exp(f(x_1, x'_2, \\dots, x_d))}.
+        f(c) = {\\exp(c) \\over \\sum_{c_d} \\exp(c_d)}.
```
Comment (delta2323) [inline chainer/functions/activation/log_softmax.py]: Suggestion: `\\sum_{d}` instead of `\\sum_{c_d}` is natural to me.

Comment (keisuke-umezawa) [inline chainer/functions/activation/log_softmax.py]: I think so too. I fixed it.

Comment (delta2323) [inline chainer/functions/activation/log_softmax.py]: Did you fix it already? It seems the change is not applied.

**Comment by keisuke-umezawa** [comment]
@delta2323 
I'm sorry. I forget to push the commit...

**Comment by delta2323** [comment]
LGTM
